controllers/admin/AdminImportController.php

```diff
@@ -3682,30 +3697,49 @@ public function storeContactImport($offset = false, $limit = false, $validateOnl
             );
         }
         $this->closeCsvFile($handle);
+
+        return $line_count;
     }
 
-    public function storeContactImportOne($info, $shop_is_feature_active, $regenerate, $force_ids, $validateOnly = false)
+    /**
+     * Import a store.
+     * If passed data contains a known store id, this store will be updated.
+     * Can also be used for store data validation only.
+     *
+     * @param array $info         : Store data
+     * @param       $notUsed      : Not used anymore.
+     * @param bool  $regenerate   : If images should be regenerated
+     * @param bool  $forceIds     : If passed store id should be used for store creation
+     * @param bool  $validateOnly : If set to true, store will not be updated nor created with $info data
+     *
+     * @return void
+     */
+    public function storeContactImportOne($info, $notUsed, $regenerate, $forceIds, $validateOnly = false)
     {
         AdminImportController::setDefaultValues($info);
 
-        if ($force_ids && isset($info['id']) && (int)$info['id']) {
-            $store = new Store((int)$info['id']);
-        } else {
-            if (array_key_exists('id', $info) && (int)$info['id'] && Store::existsInDatabase((int)$info['id'], 'store')) {
-                $store = new Store((int)$info['id']);
-            } else {
-                $store = new Store();
-            }
-        }
+        $storeId = isset($info['id']) ? (int)$info['id'] : null;
+        $store   = new Store($storeId);
+
+        $store->force_id = (bool)$forceIds;
```

Comment:
What happen for `$force_ids` and `Store::existsInDatabase(...)`?

You should explain why you changed the tests, because after a quick look this is unclear. If we want to refacto it without modifying the test cases, I would have written i.e:

```php
if (!empty($info['id']) && ($force_ids || Store::existsInDatabase((int)$info['id'], 'store'))) {
       $storeId = (int)$info['id'];
} else {
       $storeId = null;
}
```